ochazuke/webhooks/helpers.py

```diff
@@ -0,0 +1,238 @@
+    if action == 'opened':
+        add_new_issue(info)
+    elif action == 'edited':
+        issue_header_edit(info)
+    elif action == ('closed' or 'reopened'):
+        issue_status_change(info)
+    elif action == ('milestoned' or 'unmilestoned'):
+        issue_milestone_change(info)
+    elif action == ('labeled' or 'unlabeled'):
+        issue_label_change(info)
+    # Store all new events except assigned/unassigned (filtered out in route)
+    add_new_event(info)
+
+
+def add_new_issue(info):
+    """Create an issue object to insert into db from an 'opened' issue event.
+
+    When a new issue is opened, we insert it into our issue table, including:
+    - github number (int, 'id')
+    - title (text, 'header')
+    - status (boolean, 'is_open')
+    - creation timestamp ('created_at')
+    - milestone id number (int, 'milestone_id')
+    """
+    milestone_title = info.get('milestone')
+    milestone = Milestone.query.filter_by(milestone_title).one()
+    bug = Issue(info.get('issue_id'), info.get('header'),
+                info.get('created_at'), milestone_id=milestone.id)
+    # TODO: log failures/errors as well?
+    # Add issue to staging
+    db.session.add(bug)
+    # Perform the actual insertion to the database
+    db.session.commit()
+    log = app.logger
+    log.setLevel(logging.INFO)
+    msg = 'New issue ({iss}) successfully added to database.'.format(iss=bug)
+    log.info(msg)
+
+
+def add_new_event(info):
+    """Create an event object to insert into db from a new issue event.
+
+    When a new event is signaled, we insert it into the event table, including:
+    - github issue number (int, 'issue_id')
+    - username of user who triggered the event (text, 'actor')
+    - what the event was (text, 'action')
+    - any relevant details (json, 'details') -- see models.Event
+    - when we received the event (timestamp, 'received_at')
+    We assign each event a unique id automatically upon insertion to the db.
+    """
+    event = Event(info.get('issue_id'), info.get('actor'), info.get('action'),
+                  info.get('details'), info.get('received_at'))
+    # TODO: log failures/errors as well?
+    # Add event to staging
+    db.session.add(event)
+    # Insert event into event table
+    db.session.commit()
+    log = app.logger
+    log.setLevel(logging.INFO)
+    msg = 'New event ({evt}) successfully added to database.'.format(evt=event)
+    log.info(msg)
+
+
+def issue_header_edit(info):
+    """Update issue table with edited header text."""
+    # Fetch existing issue from issue table
+    bug = Issue.query.get(info.get('issue_id'))
+    # Update header and commit changes
+    bug.header = info.get('header')
+    db.session.commit()
+
+
+def issue_status_change(info):
+    """Toggle an issue's 'is_open' status in table between true and false."""
+    bug = Issue.query.get(info.get('issue_id'))
+    bug.is_open = (False if bug.is_open else True)
```

Comment:
`bug.is_open` is already a boolean in the DB.

the `info` here is `issue_event_info` if I followed correctly the code and this is called on the action `closed` or `reopened`.

Maybe the code could use directly the `action` (real information) instead of flipping on a call to the function.

```python
def issue_status_change(info, action):
    """Toggle an issue's 'is_open' status in table between true and false."""
    bug = Issue.query.get(info.get('issue_id'))
    status = {'closed': False, 'reopened': True}
    bug.is_open = status[action]
    db.session.commit()
```

Reply:
Fancy! That could definitely eliminate errors on accidentally firing an update twice or some other unpredictable event. 👍